log4j2-ecs-layout/src/test/java/co/elastic/logging/log4j2/IntegrationTest.java

```diff
@@ -0,0 +1,35 @@
+package co.elastic.logging.log4j2;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.core.LoggerContext;
+import org.apache.logging.log4j.core.config.ConfigurationSource;
+import org.apache.logging.log4j.core.config.Configurator;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+public class IntegrationTest {
+
+    @BeforeEach
+    public void init() throws IOException {
+//        ConfigurationSource source = new ConfigurationSource(new FileInputStream("log4j2.properties"));
+        Configurator.initialize(null, "log4j2.properties");
+    }
+
+    @Test
+    public void testxxx(){
+        Logger logger = LogManager.getLogger(getClass());
+        LoggerContext.getContext().getConfiguration().getProperties().put("node.id", "foo");
```

Comment:
This is actually the line which is wrong. It works when setting the `currentContext` parameter to false.

```suggestion
        LoggerContext.getContext(false).getConfiguration().getProperties().put("node.id", "foo");
```

If that's feasible, it would be much simpler and also more performant to not support pattern layouts within `EcsLayout`. Pattern layouts are meant to be used for the whole message. Using them for an alternative to lookups would be kind of a hack.

Reply:
I think that we should still support pattern converters as they would be a great way to extend the ECSLayout. I understand that the ECS is expecting some fields formatted in one way, but at the same time apps using it have requirements and user's expectation that we want to maintain.

Reply:
Performance wise I think there should not be a difference. we can always measure this.

Reply:
In contrast to other libraries, this one is opinionated towards ECS. If users want to customize the field name where the logger name is stored, this is probably not the library for them. I see ECS as a protocol which this library conforms to.

Also, just by providing the ability to resolve an additional property like this
```xml
    <KeyValuePair key="logger" value="%logger"/>
```

Does not remove the default `log.logger` ECS field.

On the other hand, it doesn't do much harm and it's easier to integrate than I have originally thought. So I went ahead and created a PR for it: https://github.com/elastic/java-ecs-logging/pull/24

I would still suggest relying on lookups as opposed to pattern converters in Elasticsearch as lookups are more widely supported by various other libraries. Also note, that something like the `%node_and_cluster_id` or `%exceptionAsJson` pattern would not work here, as the value will always be JSON encoded.

Reply:
at the same time ECS is a permissive schema, we will definitely have fields which are not defined there.
If elasticsearch is to be migrated to ECS Layout then obviously we want to make sure all the current use cases work. I would prefer to avoid maintaining 3 layouts at the same time. (we still maintain plain text logs)
I am worried that `%exceptionAsJson` would not work. Lots of users or developers still rely on visually checking logs and having an unreadable exception + stacktrace would be bad experience for them.

Reply:
I've added a PR to serialize the `stackTraceAsArray`: https://github.com/elastic/java-ecs-logging/pull/28

Reply:
I've added a PR to serialize the stackTraceAsArray: #28